### ISSUE TITLE
(keveman): Eigen on GPU dispatches to an optimized implementation 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#4054]
for an expression of the form lhs = rhs.sum().
lhs = -rhs.sum() doesn't match the above pattern, so folding in the
negation before calling sum().
 sum(-labels *
    ((logits - max_logits) - log(sum(exp(logits - max_logits)))))
 along classes

----
*opened via [imdone.io](https://imdone.io) from a code comment on [59f1eba5fb94506a205fa2e81145667754739da5](https://github.com/imdone/tensorflow/commit/59f1eba5fb94506a205fa2e81145667754739da5) by Martin Wicke wicke@google.com*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/core/kernels/xent_op.h#L106-L117
